hudi-flink/src/main/java/org/apache/hudi/sink/bootstrap/BootstrapOperator.java

```diff
@@ -151,11 +151,12 @@ protected void preLoadIndexRecords() throws Exception {
    */
   private void waitForBootstrapReady(int taskID) {
     int taskNum = getRuntimeContext().getNumberOfParallelSubtasks();
+    int attemptNum = getRuntimeContext().getAttemptNumber();
     int readyTaskNum = 1;
     while (taskNum != readyTaskNum) {
       try {
-        readyTaskNum = aggregateManager.updateGlobalAggregate(BootstrapAggFunction.NAME, taskID, new BootstrapAggFunction());
-        LOG.info("Waiting for other bootstrap tasks to complete, taskId = {}.", taskID);
+        readyTaskNum = aggregateManager.updateGlobalAggregate(BootstrapAggFunction.NAME + "_" + attemptNum, taskID, new BootstrapAggFunction());
+        LOG.info("Waiting for other bootstrap tasks to complete, taskId = {}, attemptNum = {}.", taskID, attemptNum);
```

Comment:
Hello, can you explain why we need this change ?

Reply:
Ok,once flink job with index.bootstrap=true failed like taskmanager lost
if the job restart with the same GlobalAggregate name, it will reuse the `accumulators` in JobMaster
and then, some parallelism of BootstrapOperator that faster then others will send records downstream
without wait for all bootstrap task done

Reply:
Only when the accumulator received all the task bootstrap info, the `readyTaskNum` matches and returns true, does that work for your case ? Because the fail over retry does not increase the `readyTaskNum` right ?

Reply:
yes, you are right